src/test-support/FakeKuma.ts

```diff
@@ -294,6 +295,43 @@ export class KumaModule {
       supportedBackends,
     }
   }
+
+  ruleMatch({ kind }: { kind?: 'path' | 'method' | 'headers' | 'queryParams' } = { kind: 'path' }): ToTargetRefRuleMatch {
+    const _kind = kind ?? this.faker.helpers.arrayElement<'path' | 'method' | 'headers' | 'queryParams'>(['path', 'method', 'headers', 'queryParams'])
+
+    return {
+      ...(_kind === 'path' && {
+        path: {
+          value: '/api',
+          type: 'PathPrefix',
+        },
+      }),
+      ...(_kind === 'method' && {
+        method: this.faker.internet.httpMethod(),
+      }),
+      ...(_kind === 'queryParams' && {
+        queryParams: [
+          {
+            name: 'size',
+            type: 'Exact',
+            value: '1',
+          },
+        ],
+      }),
+      ...(_kind === 'headers' && {
+        headers: [
+          {
+            name: 'X-Test-Header',
+            value: 'test-1',
+          },
+          {
+            name: 'X-Test-Header',
+            value: 'test-2',
+          },
+        ],
+      }),
+    }
+  }
```

Comment:
I'd like to try to avoid having to many 'heavy' things on the fake kuma object. Smaller things aren't so bade (like `tags` for example). Here something like `kuma.httpHeaders` etc would be better than the full 'heavier' ruleMatch object.

Similarly to our actual application frontend templates, our mock templates should be made up of these smaller 'components' to give more flexibility/visibility - sometimes a bit of repetition is an ok trade for flexibility/visibility.

Tinier thing which might be a part-consequence of the above is that I'm not entirely sure whether we'd ever get two headers with the same name in an API response - not sure what would happen in real kuma if you tried to do that.

Anyway, bit of a comment more than anything, not a hard request unless you wanted to change something here yourself.